AK/String.cpp

```diff
@@ -469,18 +462,18 @@ bool String::equals_ignoring_ascii_case(StringView other) const
 String String::repeated(String const& input, size_t count)
 {
     VERIFY(!Checked<size_t>::multiplication_would_overflow(count, input.bytes().size()));
-    u8* buffer = nullptr;
-    auto data = MUST(Detail::StringData::create_uninitialized(count * input.bytes().size(), buffer));
 
-    if (input.bytes().size() == 1) {
-        memset(buffer, input.bytes().first(), count);
-        return String { move(data) };
-    }
-
-    for (size_t i = 0; i < count; ++i) {
-        memcpy(buffer + (i * input.bytes().size()), input.bytes().data(), input.bytes().size());
-    }
-    return String { data };
+    String result;
+    MUST(result.replace_with_new_string(count * input.bytes().size(), [&](u8* buffer) {
+        if (input.bytes().size() == 1) {
+            memset(buffer, input.bytes()[0], count);
```

Comment:
first, this is making the refactored code less readable, since .first is more readable than [0]. and second, we have `Span::fill`, so I think you could just write.

```suggestion
            Bytes { buffer, count }.fill(input.bytes().first());
```